flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/failover/flip1/RestartPipelinedRegionFailoverStrategyTest.java

```diff
@@ -369,35 +252,85 @@ public void testRegionFailoverForMultipleVerticesRegions() throws Exception {
-					new Exception("Test failure"))));
+		verifyThatFailedExecution(strategy, v3).restarts(v3, v4, v5, v6);
+
+		TestingSchedulingResultPartition v2out = v3.getConsumedResults().iterator().next();
+		verifyThatFailedExecution(strategy, v3)
+				.partitionConnectionCause(v2out)
+				.restarts(v1, v2, v3, v4, v5, v6);
 	}
 
-	// ------------------------------------------------------------------------
-	//  utilities
-	// ------------------------------------------------------------------------
+	/**
+	 * Tests region failover does not restart vertexes which are already in initial CREATED state.
+	 * <pre>
+	 *     (v1) ---> (v2) --|--> (v3) ---> (v4) --|--> (v5) ---> (v6)
+	 *
+	 *           ^          ^          ^          ^          ^
+	 *           |          |          |          |          |
+	 *     (pipelined) (blocking) (pipelined) (blocking) (pipelined)
+	 * </pre>
+	 * Component 1: 1,2; component 2: 3,4; component 3: 5,6
+	 */
+	@Test
+	public void testRegionFailoverDoesNotRestartCreatedExecutions() {
+		TestingSchedulingTopology topology = new TestingSchedulingTopology();
+
+		TestingSchedulingExecutionVertex v1 = topology.newExecutionVertex(ExecutionState.CREATED);
+		TestingSchedulingExecutionVertex v2 = topology.newExecutionVertex(ExecutionState.CREATED);
+		TestingSchedulingExecutionVertex v3 = topology.newExecutionVertex(ExecutionState.CREATED);
+		TestingSchedulingExecutionVertex v4 = topology.newExecutionVertex(ExecutionState.CREATED);
+		TestingSchedulingExecutionVertex v5 = topology.newExecutionVertex(ExecutionState.CREATED);
+		TestingSchedulingExecutionVertex v6 = topology.newExecutionVertex(ExecutionState.CREATED);
+
+		topology.connect(v1, v2, ResultPartitionType.PIPELINED);
+		topology.connect(v2, v3, ResultPartitionType.BLOCKING);
+		topology.connect(v3, v4, ResultPartitionType.PIPELINED);
+		topology.connect(v4, v5, ResultPartitionType.BLOCKING);
+		topology.connect(v5, v6, ResultPartitionType.PIPELINED);
+
+		RestartPipelinedRegionFailoverStrategy strategy = new RestartPipelinedRegionFailoverStrategy(topology);
+
+		verifyThatFailedExecution(strategy, v3).restarts();
+		TestingSchedulingResultPartition v2out = v3.getConsumedResults().iterator().next();
+		verifyThatFailedExecution(strategy, v3).partitionConnectionCause(v2out).restarts();
+	}
+
+	private static VerificationContext verifyThatFailedExecution(
+			FailoverStrategy strategy,
+			SchedulingExecutionVertex executionVertex) {
+		return new VerificationContext(strategy, executionVertex);
+	}
+
+	private static class VerificationContext {
+		private final FailoverStrategy strategy;
+		private final SchedulingExecutionVertex executionVertex;
+		private Throwable cause = new Exception("Test failure");
+
+		private VerificationContext(
+				FailoverStrategy strategy,
+				SchedulingExecutionVertex executionVertex) {
+			this.strategy = strategy;
+			this.executionVertex = executionVertex;
+		}
+
+		private VerificationContext partitionConnectionCause(
```

Comment:
```suggestion
		private VerificationContext withPartitionConnectionCause(
```

Reply:
Do we even need this method at all? We could just overload `verifyThatFailedExecution`.

Reply:
true, but I think this improves readability